solidity/contracts/bridge/VendingMachine.sol

```diff
@@ -0,0 +1,250 @@
+    }
+
+    /// @notice Allows the Governance to withdraw unmint fees accumulated by
+    ///         `VendingMachine`.
+    /// @param recipient The address receiving the fees
+    /// @param amount The amount of fees in TBTC v2 to withdraw
+    function withdrawFees(address recipient, uint256 amount)
+        external
+        onlyOwner
+    {
+        tbtcV2.safeTransfer(recipient, amount);
+    }
+
+    /// @notice Allows the Governance to begin unmint fee update process.
+    ///         The update process needs to be finalized with a call to
+    ///         `finalizeUnmintFeeUpdate` function after the `GOVERNANCE_DELAY`
+    ///         passes.
+    /// @param _newUnmintFee The new unmint fee
+    function beginUnmintFeeUpdate(uint256 _newUnmintFee) external onlyOwner {
+        /* solhint-disable-next-line not-rely-on-time */
+        emit UnmintFeeUpdateStarted(_newUnmintFee, block.timestamp);
+        newUnmintFee = _newUnmintFee;
+        /* solhint-disable-next-line not-rely-on-time */
+        unmintFeeChangeInitiated = block.timestamp;
+    }
+
+    /// @notice Allows the Governance to finalize unmint fee update process.
+    ///         The update process needs to be first initiated with a call to
+    ///         `beginUnmintFeeUpdate` and the `GOVERNANCE_DELAY` needs to pass.
+    function finalizeUnmintFeeUpdate()
+        external
+        onlyOwner
+        onlyAfterGovernanceDelay(unmintFeeChangeInitiated)
+    {
+        emit UnmintFeeUpdated(newUnmintFee);
+        unmintFee = newUnmintFee;
+        newUnmintFee = 0;
+        unmintFeeChangeInitiated = 0;
+    }
+
+    /// @notice Allows the Governance to begin vending machine update process.
+    ///         The update process needs to be finalized with a call to
+    ///         `finalizeVendingMachineUpdate` function after the
+    ///         `GOVERNANCE_DELAY` passes.
+    /// @param _newVendingMachine The new vending machine address
+    function beginVendingMachineUpdate(address _newVendingMachine)
+        external
+        onlyOwner
+    {
+        require(
+            _newVendingMachine != address(0),
+            "New VendingMachine can not be zero address"
+        );
+
+        /* solhint-disable-next-line not-rely-on-time */
+        emit VendingMachineUpdateStarted(_newVendingMachine, block.timestamp);
+        newVendingMachine = _newVendingMachine;
+        /* solhint-disable-next-line not-rely-on-time */
+        vendingMachineUpdateInitiated = block.timestamp;
+    }
+
+    /// @notice Allows the Governance to finalize vending machine update process.
+    ///         The update process needs to be first initiated with a call to
+    ///         `beginVendingMachineUpdate` and the `GOVERNANCE_DELAY` needs to
+    ///         pass. Once the update is finalized, the new vending machine will
+    ///         become an owner of TBTC v2 token.
+    function finalizeVendingMachineUpdate()
+        external
+        onlyOwner
```

Comment:
Is the `onlyOwner` modifier necessary here? Can't the finalization be callable by anyone?

Reply:
All `finalize*` calls in tBTC are `onlyOwner`, same for coverage pools so I decided to follow the same pattern here.